libsolidity/formal/SMTPortfolio.cpp

```diff
@@ -32,42 +32,42 @@ using namespace dev::solidity::smt;
 
 SMTPortfolio::SMTPortfolio(map<h256, string> const& _smtlib2Responses)
 {
-	m_solvers.emplace_back(make_shared<smt::SMTLib2Interface>(_smtlib2Responses));
+	m_solvers.emplace_back(make_unique<smt::SMTLib2Interface>(_smtlib2Responses));
 #ifdef HAVE_Z3
-	m_solvers.emplace_back(make_shared<smt::Z3Interface>());
+	m_solvers.emplace_back(make_unique<smt::Z3Interface>());
 #endif
 #ifdef HAVE_CVC4
-	m_solvers.emplace_back(make_shared<smt::CVC4Interface>());
+	m_solvers.emplace_back(make_unique<smt::CVC4Interface>());
 #endif
 }
 
 void SMTPortfolio::reset()
 {
-	for (auto s : m_solvers)
+	for (auto const& s : m_solvers)
```

Comment:
Do we usually have a space before the ``:`` in range based fors? I think we don't and since you're touching those anyways ;-)...

Reply:
we don't :p